### PR TITLE
fixed bug in tz_localize to throw an error if the argument to the ambiguous parameter is not one of the 4 correct options

### LINKED ISSUE #49565: BUG: when `floor`ing, `ambiguous` parameter unnecessarily used (and raising Error)
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd

ts = pd.Timestamp("2020-10-25 00:00", tz='Europe/Berlin')
# This works:
ts.floor('H') # Timestamp('2020-10-25 00:00:00+0200', tz='Europe/Berlin')
# But this gives an error:
ts.floor('H', ambiguous='infer') # ValueError: Cannot infer offset with only one time.
```

### Issue Description

The value of the `ambiguous` parameter is used by the method, even if there is no ambiguity.

### Expected Behavior

Both calls to the `.floor` method should give the same outcome.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 91111fd99898d9dcaa6bf6bedb662db4108da6e6
python           : 3.9.13.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19045
machine          : AMD64
processor        : Intel64 Family 6 Model 165 Stepping 2, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : de_DE.cp1252

pandas           : 1.5.1
numpy            : 1.23.4
pytz             : 2022.6
dateutil         : 2.8.2
setuptools       : 63.4.1
pip              : 22.2.2
Cython           : None
pytest           : 7.1.3
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : 8.5.0
pandas_datareader: None
bs4              : None
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : 3.6.2
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : 3.0.10
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : None
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None
tzdata           : None

</details>

### PR DESCRIPTION
**Pull request by jcaoun**
- [x] closes #49565 
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by mroeschke — CHANGES_REQUESTED** [review]
Given https://github.com/pandas-dev/pandas/issues/49565#issuecomment-1325468574, I do believe the original issue is behaving as expected and a fix isn't needed here.

If you are looking for issues to tackle, I would recommend searching for issues that have `good first issue` and avoid issues that still have `need triage`

File: floor_timestamp_tests.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import pandas as pd
+import unittest
+
+class MyTestCase(unittest.TestCase):
```
Comment (MarcoGorelli) [inline floor_timestamp_tests.py:4]: this isn't the right location for tests, can you search for similar tests within the pandas test suite and add it there?

Also, you can use `pytest.mark.parametrize` to parametrize over arguments, no need to write 6 different tests

File: pandas/_libs/tslibs/timestamps.pyx [hunk 2]
```diff
@@ -1926,8 +1926,9 @@ default 'raise'
         >>> pd.NaT.tz_localize()
         NaT
         """
-        if ambiguous == 'infer':
-            raise ValueError('Cannot infer offset with only one time.')
+
+        if ambiguous != True and ambiguous != False and ambiguous != 'NaT' and ambiguous != 'raise':
```
Comment (MarcoGorelli) [inline pandas/_libs/tslibs/timestamps.pyx:1930]: let's write this as 
```python
if not isinstance(ambiguous, bool) and ambiguous not in {'NaT', 'raise'}:
```

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

**Review by jcaoun — COMMENTED** [review]
I modified timestamps.pyx to include the changes you suggested, as well as modified test_timezones.py and made sure those test cases passed

File: floor_timestamp_tests.py (hunk 1 above)
Comment (MarcoGorelli) [reply floor_timestamp_tests.py:4]: this comment still needs addressing

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
thanks for updating!

this is failing a lot of tests, could you take a lot please?

File: floor_timestamp_tests.py (hunk 1 above)
Comment (jcaoun) [reply floor_timestamp_tests.py:4]: You can disregard floor_timestamp_tests.py. The test cases in pandas/tests/scalar/timestamp/test_timezones.py cover the bug already and they are passing on my end

Comment (MarcoGorelli) [reply floor_timestamp_tests.py:4]: ok - will you remove it?

**Comment by jcaoun** [comment]
I believe the test cases were failing after I merged with the main branch to keep it up to date, but I fixed the issue and they should be passing now

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
can you run the pre-commit checks and undo having commented-out lines please?

File: pandas/tests/scalar/timestamp/test_timezones.py [hunk 3]
```diff
@@ -141,9 +124,9 @@ def test_tz_localize_ambiguous_raise(self):
         with pytest.raises(AmbiguousTimeError, match=msg):
             ts.tz_localize("US/Pacific", ambiguous="raise")
 
-    def test_tz_localize_nonexistent_invalid_arg(self, warsaw):
+    def test_tz_localize_nonexistent_invalid_arg(self):
         # GH 22644
-        tz = warsaw
+        tz = "Europe/Warsaw"
```
Comment (MarcoGorelli) [inline pandas/tests/scalar/timestamp/test_timezones.py:129]: why does this need to change?

File: pandas/tests/scalar/timestamp/test_timezones.py [hunk 4]
```diff
@@ -75,19 +69,6 @@ def test_tz_localize_ambiguous_bool(self, unit):
...
-        if ZoneInfo is not None:
-            try:
-                tz = ZoneInfo("US/Central")
-            except KeyError:
-                # no tzdata
-                pass
-            else:
-                with pytest.raises(pytz.AmbiguousTimeError, match=msg):
-                    ts.tz_localize(tz)
-
```
Comment (MarcoGorelli) [inline pandas/tests/scalar/timestamp/test_timezones.py:90]: why delete this?

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

File: pandas/tests/scalar/timestamp/test_timezones.py (hunk 3 above)
Comment (jcaoun) [reply pandas/tests/scalar/timestamp/test_timezones.py:129]: I didn't change that so I'm not sure, it changed when I merged with the main branch

Comment (MarcoGorelli) [reply pandas/tests/scalar/timestamp/test_timezones.py:129]: looks like something went wrong when merging, please try again

File: pandas/tests/scalar/timestamp/test_timezones.py (hunk 4 above)
Comment (jcaoun) [reply pandas/tests/scalar/timestamp/test_timezones.py:90]: This also got deleted when I merged with the main branch

File: pandas/tests/scalar/timestamp/test_timezones.py (hunk 3 above)
Comment (jcaoun) [reply pandas/tests/scalar/timestamp/test_timezones.py:129]: It looks like there were some changes to these files on the pandas-dev main branch, I'll sync with my branch to get it up to date first

Comment (jcaoun) [reply pandas/tests/scalar/timestamp/test_timezones.py]: Ok, so I merged with with the pandas/dev main branch, then re-added my changed with tests and pre-commit passing. I accidentally closed this while merging so if you can't re-open, the new pull request is here: https://github.com/pandas-dev/pandas/pull/50024

**Pull request closed without merging**